_posts/2016-05-13-six-steps-to-detect-session-issues.md

```diff
@@ -0,0 +1,380 @@
+---
+layout: post
+title:  "6 steps to investigate unexpected logouts in your Java web application"
+date:   2016-05-13 08:00:00 +0200
+author: "Kamil Kubacki"
+tags: web java springmvc session session-management
+---
+
+
+I would like to share with you six steps that may help in investigation of problems
+related to wrong session management and/or wrong infrastructure configuration for your
+web applications, especially Java web applications.
+
+# Introduction
+The steps were prepared based on the real experience with finding
+out the root cause of unexpected users' logouts in my Spring MVC web application.
+Users' claimed that they were being logged out during continuous usage of
+application (e.g. during clicking through menu). I was trying to reproduce
+those unexpected logouts on my own but it resulted in failure. First idea about
+possible cause of logouts led me to thinking that users were logged out due to
+exceeded idle timeout available. My assumptions were denied by clients (and by me)
+so I started looking for the help in the Internet. I was not able to find a clear
+way to deal with those kind of problems so I prepared a plan to figure out the solution.
+The plan consists of six steps and this blog post represents it.
+
+# Application and infrastructure (overview)
+The application I am writing about is a Spring MVC based app with support
+of Spring Security for authentication and authorization management.
+The application is deployed into multiple [JBoss AS](http://jbossas.jboss.org/)
+instances on multiple machines. Each JBoss instance is balanced by
+[Apache HTTP Server](https://httpd.apache.org/) and each of the machines is
+balanced by load balancer. To give you better view at the infrastructure
+please take a look at the picture below.
+
+<img src="/images/blog/posts/six-steps-to-detect-session-issues/architecture.png" alt="infrastructure" />
+
+
+# The Six Steps
+
+At this point, you are familiar with technical details regarding application
+so it is high time to look into the plan.
+
+# <b>1. Verify your application's configuration.</b>
+It is a common think for web application to has ability to configure session management.
+The variety of configuration is based on the web framework you decide to use.
+As for Spring MVC you can influence session management with ```<session-config>...</session-config>```
+tag inside web.xml. One of the first things that I checked during investigation
+was session timeout configuration under this tag. Session timeout property is
+responsible for invalidating users' sessions if they exceed the defined idle time
```

Comment:
idle timeout (defined in minutes) 
and you can remove "The value of this property is defined in minutes." sentence to prevent repeating "this property"